fvm/src/executor/default.rs

```diff
@@ -149,12 +153,15 @@ where
         };
 
         match apply_kind {
-            ApplyKind::Explicit => self.finish_message(msg, receipt, failure_info, gas_cost),
+            ApplyKind::Explicit => {
+                self.finish_message(msg, receipt, failure_info, gas_cost, exec_trace)
```

Comment:
Hm, this was confusing to me. I didn't expect `finish_message` to operate on the `exec_trace`, so I didn't understand why we passed it in. And indeed it doesn't.

Did you consider:

```rust
.map(|apply_ret| {
    apply_ret.exec_trace = exec_trace;
    apply_ret
})
```

It's a few more lines of code but it's less surprising (for me). Alternatively we could use a one-liner:

```rust
.inspect(|apply_ret| { apply_ret.exec_trace = exec_trace })
```

But it requires us to enable an experimental API: https://github.com/rust-lang/rust/issues/91345.